File: src/praktikum2/kegiatan1/Hewan.java
package praktikum2.kegiatan1;

import java.util.ArrayList;
import java.util.List;

public class Hewan<T> {
    ArrayList<String> namaHewan = new ArrayList<>(List.of("Angsa", "Bebek", "Cicak", "Domba", "Elang", "Gajah"));

    public static void main(String[] args) {
        Hewan<String> hewan = new Hewan<String>();

        System.out.println("=== Tahap 1 ===");
        System.out.println("Nama Hewan: " + hewan.namaHewan);

        // tahap 2
        System.out.println("\n=== Tahap 2 ===");
        hewan.namaHewan.add("Badak");
        hewan.namaHewan.add("Bebek");

        int totalBebek = 0;
        for (String h : hewan.namaHewan) {
            if (h.equals("Bebek")) {
                totalBebek++;
            }
        }

        System.out.println("Total Bebek: " + totalBebek);
        System.out.println(
                "Posisi Index Bebek: " + hewan.namaHewan.indexOf("Bebek") + " dan "
                        + hewan.namaHewan.lastIndexOf("Bebek"));

        System.out.println("Nama Hewan: " + hewan.namaHewan);

        // tahap 3
        System.out.println("\n=== Tahap 3 ===");
        hewan.namaHewan.remove(hewan.namaHewan.indexOf("Bebek"));

        System.out.println("Nama Hewan: " + hewan.namaHewan);

        // tahap 4
        System.out.println("\n=== Tahap 4 ===");
        System.out.println("Index ke-0: " + hewan.namaHewan.get(0));
        System.out.println("Index ke-2: " + hewan.namaHewan.get(2));
        hewan.namaHewan.remove(0);
        System.out.println("Nama Hewan: " + hewan.namaHewan);

        // tahap 5
        System.out.println("\n=== Tahap 5 ===");
        System.out.println("Nama Hewan: " + hewan.namaHewan);

        hewan.namaHewan.set(0, "Ular");
        hewan.namaHewan.add(2, "Itik");

        System.out.println("Nama Hewan: " + hewan.namaHewan);

        // tahap 6
        System.out.println("\n=== Tahap 6 ===");
        hewan.namaHewan.subList(2, 5).clear();

        System.out.println("Nama Hewan: " + hewan.namaHewan);

        // tahap 7
        System.out.println("\n=== Tahap 7 ===");
        System.out.println("Elemen pertama  : " + hewan.namaHewan.get(0));
        System.out.println("Elemen terakhir : " + hewan.namaHewan.get(hewan.namaHewan.size() - 1));

        // tahap 8
        System.out.println("\n=== Tahap 8 ===");
        System.out.println("Total elemen hewan : " + hewan.namaHewan.size());

        // tahap 9
        System.out.println("\n=== Tahap 9 ===");
        System.out.println("Posisi index badak : " + hewan.namaHewan.indexOf("Badak"));
    }
}

File: src/praktikum2/kegiatan1/Hewan.py
__author__      = "rizkyhaksono"
__copyright__   = "Copyright 2023, Malang"

class Hewan:
    # tahap 1
    namaHewan = ["Angsa", "Bebek", "Cicak", "Domba", "Elang", "Gajah"]

    print("=== Tahap 1 ===")
    print("Nama Hewan : ", namaHewan)

    # tahap 2
    print("\n=== Tahap 2 ===")
    namaHewan.append("Badak")
    namaHewan.append("Bebek")

    totalBebek = 0
    for (i, nama) in enumerate(namaHewan):
        if nama == "Bebek":
            totalBebek += 1

    print("Total Bebek : ", totalBebek)
    print("Posisi Index Bebek : ", [i for i, x in enumerate(namaHewan) if x == "Bebek"])

    print("Nama Hewan : ", namaHewan)

    # tahap 3
    print("\n=== Tahap 3 ===")
    namaHewan.remove("Bebek")

    print("Nama Hewan : ", namaHewan)

    # tahap 4
    print("\n=== Tahap 4 ===")
    print("Index ke-0 : ", namaHewan[0])
    print("Index ke-2 : ", namaHewan[2])
    print("Nama Hewan : ", namaHewan)

    # tahap 5
    namaHewan2 = ["Cicak", "Domba", "Elang", "Gajah", "Badak", "Bebek"]
    print("\n=== Tahap 5 ===")
    print("Nama Hewan : ", namaHewan2)

    namaHewan2[0] = "Ular"
    namaHewan2.insert(2, "Itik")

    print("Nama Hewan : ", namaHewan2)

    # tahap 6
    print("\n=== Tahap 6 ===")
    del namaHewan2[2:5]

    print("Nama Hewan : ", namaHewan2)

    # tahap 7
    print("\n=== Tahap 7 ===")
    print("Elemen pertama : ", namaHewan2[0])
    print("Elemen terakhir : ", namaHewan2[-1])

    # tahap 8
    print("\n=== Tahap 8 ===")
    print("Total elemen hewan 1 : ", len(namaHewan))
    print("Total elemen hewan 2 : ", len(namaHewan2))

    # tahap 9
    print("\n=== Tahap 9 ===")
    print("Posisi index badak : ", namaHewan.index("Badak"))